### ISSUE TITLE
German language fallback

### ISSUE THREAD

**Issue by mbajur** [spree-contrib/spree_i18n#546]
Hello
i'm encountering a really strange issue with spree_i18n. Basically, with spree_i18n disabled everything works good. However, after enabling it and configuring two enabled languages (which are pl_PL and en_US), some kind of random frontend elements are translated to german language. It's really awkward because i have never configured that language anywhere and it's not enabled in the admin panel so i have no idea where that comes from.

Some of the parts translated to german:
* Search form in header
* Shopping cart link in header
* Shopping card
* probably a lot of others i don't know about right now

My root rails app `config.i18n.default_locale` is set to `:en`. Please note that it happens only after removing my cookies (so i assume that on that point spree and spree_i18n does not uses any enabled languages, doesn't know about my previous choices in language picker and uses some kind of default/fallback language). If i change the language in the picker, all the translations starts to work good.

Any ideas? Thanks.

**Comment by JDutil** [comment]
That is odd I'm not sure what would be causing that either if your default is set to en...

**Comment by tvdeyen** [comment]
Could you please show us the content of your `Gemfile` and `config/locales` folder? Thanks

**Comment by mbajur** [comment]
Sure, here they are:

1. Gemfile: https://gist.github.com/mbajur/3090476023d73193f387
2. config/locales - empty dir

**Comment by alepore** [comment]
probably not related but the default rails `application.rb` template is like:
```ruby
# config.i18n.default_locale = :de
```

can you double check? ;)

**Comment by mbajur** [comment]
Ok guys, i have tested spree_i18n on fresh rails/spree install and there is no such issue in there. It seems that i have some really weird cache issues - german language has been cached (i've set it with `config.i18n.default_locale` to :de about two weeks ago just to check something and it stayed that way)  and there seems to be no way to refresh that + each time i want to update my css files, i need to restart the server, clean assets and precompile them. 

There is certainly something broken with my spree instance, i'll try with the fresh one. Thanks for help attempts! :)

**Comment by mbajur** [comment]
Guys, i have tested that again on fresh install and still the cart link is not being translated. The AJAX request (for localisation set to `PL`) goes to `/cart_link` instead of `/pl/cart_link` what gives me untranslated cart link which points to untranslated cart page (`/cart` instead of `/pl/cart`). It seems that `Spree.urlFor` does not takes current locale into consideration but i'm not able to verify that because i can't find that method anywhere.

**Comment by alepore** [comment]
makes sense, can you please open a new issue for this cart link problem?
this issue is about German language fallback and a bit confusing.
thanks!
